R-package/R/lgb.Booster.R

```diff
@@ -538,6 +541,57 @@ Booster <- R6::R6Class(
       return(Predictor$new(modelfile = private$handle))
     },
 
+    configure_fast_predict = function(csr = FALSE,
+                                      start_iteration = NULL,
+                                      num_iteration = NULL,
+                                      rawscore = FALSE,
+                                      predleaf = FALSE,
+                                      predcontrib = FALSE,
+                                      params = list()) {
+
+      self$restore_handle()
+      ncols <- .Call(LGBM_BoosterGetNumFeature_R, private$handle)
+
+      if (is.null(num_iteration)) {
+        num_iteration <- -1L
+      }
+      if (is.null(start_iteration)) {
+        start_iteration <- 0L
+      }
+
+      if (!csr) {
+        fun <- LGBM_BoosterPredictForMatSingleRowFastInit_R
+      } else {
+        fun <- LGBM_BoosterPredictForCSRSingleRowFastInit_R
+      }
+
+      fast_handle <- .Call(
+        fun
+        , private$handle
+        , ncols
+        , rawscore
+        , predleaf
+        , predcontrib
+        , start_iteration
+        , num_iteration
+        , lgb.params2str(params = params)
+      )
+
+      self$fast_predict_config <- list(
+        handle = fast_handle
+        , csr = as.logical(csr)
+        , ncols = ncols
+        , start_iteration = start_iteration
+        , num_iteration = num_iteration
+        , rawscore = as.logical(rawscore)
+        , predleaf = as.logical(predleaf)
+        , predcontrib = as.logical(predcontrib)
```

Comment:
Why is it necessary to use `as.logical()` for these calls? I believe `rawscore`, `predleaf`, and `predcontrib` should already be logicals at this point, and that passing values like `1L` instead of `TRUE` (for example) shouldn't be supported.

Reply:
No, there's only one function that calls this method and it doesn't validate the parameters. Adding the conversions here would trigger a more informative user error if an invalid value is encountered.

Reply:
> Adding the conversions here would trigger a more informative user error

Adding conversions like this introduces support for passing "true-looking" values like the string `"TRUE"` or any non-0 number (https://github.com/microsoft/LightGBM/pull/5312#discussion_r946263496). I don't support allowing those values.

Please remove these conversions. In the spirit of keeping this configuration focused on being "fast", I think it's ok to forgo input type-checking for `rawscore`, `predleaf`, and `predcontrib`.

Reply:
Removed these checks in favor of the new `type` argument.

Reply:
ah great, that works. Thanks.